src/cargo/core/compiler/mod.rs

```diff
@@ -330,7 +330,7 @@ fn rustc(
     if hide_diagnostics_for_scrape_unit {
         output_options.show_diagnostics = false;
     }
-
+    let config_envs = build_runner.bcx.gctx.env_config()?.clone();
```

Comment:
Explicit `Arc::clone` is preferred.

```suggestion
    let config_envs = Arc::clone(&build_runner.bcx.gctx.env_config()?);
```